avocado/utils/process.py

```diff
@@ -135,6 +136,57 @@ def safe_kill(pid, signal):  # pylint: disable=W0621
         return False
 
 
+def get_parent_pid(pid):
+    """
+    Returns the parent PID for the given process
+
+    TODO: this is currently Linux specific, and needs to implement
+    similar features for other platforms.
+
+    :param pid: The PID of child process
+    :returns: The parent PID
+    :rtype: int
+    """
+    with open('/proc/%d/stat' % pid) as proc_stat:
+        parent_pid = proc_stat.read().split(b' ')[3]
```

Comment:
Why use bytes type in the split argument? A quick test on my machine worked well with string.

Reply:
I got this on Travis:

https://travis-ci.org/avocado-framework/avocado/jobs/429744345#L2085

And then changed to bytes.  Now, I'm asking myself why that happened.

Reply:
Oh, right, it's because of the `io.BytesIO` used in the selftest.

Reply:
Either way, IMO it's safe (and may be preferable) to use bytes because I don't expect that separating character (whitespace) to change in `/proc/$$/stat` and across locales.  I'll go ahead and use binary open modes to be more precise.